### ISSUE TITLE
Runas Causing "invalid directory" error on Windows.

### ISSUE THREAD

**Issue by doesitblend** [saltstack/salt#47125]
### Description of Issue/Question
When using `cmd.script` with the `runas` option I am getting an error "The current directory is invalid". 

### Setup
Install Salt 2017.7.5 on Windows 2016. 

Create a test user using `salt '<test minion>' user.add borneo <some password>`. 

### Steps to Reproduce Issue
Run the following state on the newly configured minion from above.

```
run_test_script:
  cmd.script:
    - source: salt://mytestscript.ps1
    - shell: powershell
    - args: '"this is a string"'
    - runas: {{ pillar['temp-db']['username'] }}
    - password: {{ pillar['temp-db']['password'] }}
```

[mytestscript.ps1.tar.gz](https://github.com/saltstack/salt/files/1921430/mytestscript.ps1.tar.gz)
[testme.sls.tar.gz](https://github.com/saltstack/salt/files/1921431/testme.sls.tar.gz)

### Versions Report
```
    Salt Version:
               Salt: 2017.7.5
     
    Dependency Versions:
               cffi: 1.10.0
           cherrypy: 10.2.1
           dateutil: 2.6.0
          docker-py: Not Installed
              gitdb: 2.0.3
          gitpython: 2.1.3
              ioflo: Not Installed
             Jinja2: 2.9.6
            libgit2: Not Installed
            libnacl: 1.6.1
           M2Crypto: Not Installed
               Mako: 1.0.6
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.17
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)]
       python-gnupg: 0.4.0
             PyYAML: 3.12
              PyZMQ: 16.0.2
               RAET: Not Installed
              smmap: 2.0.3
            timelib: 0.2.4
            Tornado: 4.5.1
                ZMQ: 4.1.6
     
    System Versions:
               dist:   
             locale: cp1252
            machine: AMD64
            release: 2016Server
             system: Windows
            version: 2016Server 10.0.14393  Multiprocessor Free
```

**Comment by doesitblend** [comment]
ZD-2428

This may be related to https://github.com/saltstack/salt/issues/45063 and/or https://github.com/saltstack/salt/issues/45643.

**Comment by garethgreenaway** [comment]
Maybe a permissions thing?  @twangboy @dwoz Thoughts?

**Comment by gtmanfred** [comment]
if you use runas in a state, you have to use runas_password, not password, those are state level arguments.

**Comment by doesitblend** [comment]
@gtmanfred Thanks for the clarification about `runas_password`. I have tried this with `runas_password` and still get the same error. I updated my state to the following:

```
run_test_script:
  cmd.script:
    - source: salt://origami/mytestscript.ps1
    - shell: powershell
    - args: '"this is a string"'
    - runas: {{ pillar['temp-db']['username'] }}
    - runas_password: {{ pillar['temp-db']['password'] }}
```

The following is the result from executing the above:

```
win2016-2:
----------
          ID: run_test_script
    Function: cmd.script
      Result: False
     Comment: Command 'run_test_script' run
     Started: 20:48:05.182000
    Duration: 157.0 ms
     Changes:   
              ----------
              pid:
                  2360
              retcode:
                  1
              stderr:
                  The current directory is invalid.
              stdout:

Summary for win2016-2
------------
Succeeded: 0 (changed=1)
Failed:    1
------------
Total states run:     1
Total run time: 157.000 ms
ERROR: Minions returned with non-zero exit code
```

**Comment by JonGriggs** [comment]
I'd like to chime in that I'm having the exact same problem.  If I add the user to the local Administrators group it works fine.

**Comment by rallytime** [comment]
This is fixed with #47412
